### PR TITLE
Added an option to normalize form data

### PR DESCRIPTION
**Pull request by florianv**
The normalizer is only applied when the data is decoded which is not the case with `multipart/form-data`.

Fix issue https://github.com/FriendsOfSymfony/FOSRestBundle/issues/868.

### CODE REVIEW

File: DependencyInjection/FOSRestExtension.php [hunk 1]
```diff
@@ -160,9 +160,11 @@ public function load(array $configs, ContainerBuilder $container)
...
 
             $arrayNormalizer = $config['body_listener']['array_normalizer'];
-            if (null !== $arrayNormalizer) {
-                $container->getDefinition($this->getAlias().'.body_listener')
-                    ->addMethodCall('setArrayNormalizer', array(new Reference($arrayNormalizer)));
+
+            if (null !== $arrayNormalizer['service']) {
+                $bodyListener = $container->getDefinition($this->getAlias().'.body_listener');
+                $bodyListener->addArgument(new Reference($arrayNormalizer['service']));
+                $bodyListener->addArgument($arrayNormalizer['forms']);
```
Comment (florianv) [inline DependencyInjection/FOSRestExtension.php]: Thanks for the suggestion, but `replaceArgument()` will throw an exception if the argument was not added (it only replaces it).

Comment (xabbuh) [inline DependencyInjection/FOSRestExtension.php]: Hm, actually you're right.

**Comment by lsmith77** [comment]
this just needs a rebase .. right?

**Comment by florianv** [comment]
@lsmith77 yes. The rebase has been done

**Pull request merged**